src/arrayMethodSort.js

```
@@ -4,9 +4,29 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const cmp =
+      typeof compareFunction === 'function'
```

If `compareFunction` is provided but is not a function, the native sort throws a `TypeError`. This implementation does not check for this and will silently use the default comparator, which does not match the requirements. You should add a check and throw a `TypeError` if `compareFunction` is not a function and is not `undefined`.